[OTEL-1153] Remap service.name to service in OTel logs (#20406)
DataDog/documentation: content/en/opentelemetry/otel_collector_datadog_exporter.md

    - `json_parser`: Parses JSON logs. By default, the filelog receiver converts each log line into a log record, which is the `body` of the logs' [data model][15]. Then, the `json_parser` converts the JSON body into attributes in the data model.
    - `trace_parser`: Extract the `trace_id` and `span_id` from the log to correlate logs and traces in Datadog. 

> #### Remap OTel service.name attribute to service in logs

Reviewer: ```suggestion
#### Remap OTel's `service.name` attribute to `service` for logs
```

> In certain Datadog Exporter versions, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`, and `service.name` is not in the default [service attributes][26] in Datadog log preprocessing.

Reviewer: ```suggestion
In certain Datadog Exporter versions, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.
```

> In certain Datadog Exporter versions, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.

Reviewer: ```suggestion
For Datadog Exporter versions 0.83.0 and later, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.
```

> To get the `service` field properly populated in logs, you can specify `service.name` to be the source of a log’s service by setting a [log service remapper processor][27].

Reviewer: ```suggestion
To get the `service` field correctly populated in logs, you can specify `service.name` to be the source of a log's service by setting a [log service remapper processor][27].
```

> To get the `service` field correctly populated in logs, you can specify `service.name` to be the source of a log's service by setting a [log service remapper processor][27].

Reviewer: ```suggestion
To get the `service` field correctly populated in your logs, you can specify `service.name` to be the source of a log's service by setting a [log service remapper processor][27].
```
